state/application.go

```diff
@@ -619,15 +619,8 @@ func (op *DestroyApplicationOperation) unassignBranchOps() ([]txn.Op, error) {
 }
 
 func removeResourcesOps(st *State, applicationID string) ([]txn.Op, error) {
-	persist, err := st.ResourcesPersistence()
-	if errors.IsNotSupported(err) {
-		// Nothing to see here, move along.
-		return nil, nil
-	}
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	ops, err := persist.NewRemoveResourcesOps(applicationID)
+	resources := st.Resources().(*resourcePersistence)
```

Comment:
is this guaranteed to be *resourcePersistence?

Reply:
yes, i could raise an error just in case